src/Workspaces/Core/Desktop/InternalUtilities/FilePathUtilities.cs

```diff
@@ -103,5 +103,18 @@ public static bool PathsEqual(string path1, string path2)
         {
             return string.Compare(path1, path2, StringComparison.OrdinalIgnoreCase) == 0;
         }
+
+        public static string Combine(string path1, string path2)
+        {
+            try
+            {
+                // don't throw exception when either path1 or path2 contains illegal path char
+                return Path.Combine(path1, path2);
+            }
+            catch
+            {
+                return string.Empty;
+            }
+        }
```

Comment:
Given that this method is only used in AbstractTableEntriesSnapshot.cs, just move it there.